lib/versioning/utils.js

```diff
@@ -0,0 +1,63 @@
+const VSConst = require('./constants').VersioningConstants;
+
+const masterVersionPrefixLength =
+    VSConst.VERSIONING_AFFIXES.MASTER_VERSION_PREFIX.length;
+const otherVersionPrefixLength =
+    VSConst.VERSIONING_AFFIXES.OTHER_VERSION_PREFIX.length;
+const SEP = VSConst.VERSIONING_AFFIXES.SEPARATOR;
+
+module.exports.VersioningUtils = {
+    // using explicit encode and decode functions for version object to enable
+    // future extensions, such as attatching external information to the string
+    // value from request instead of having to parse and stringify the value
+    // Example: "prefixlen_{prefix}{value_from_request}"
+    encodeVersion: version => JSON.stringify(version),
+    decodeVersion: version => {
+        try {
+            return JSON.parse(version);
+        } catch (err) {
+            // using undefined here because JSON.parse("null") === null
+            return undefined;
+            // what to do with the error
```

Comment:
I agree that we can't just do nothing about the error.  If this is being used in such a way that throwing makes more sense, then leave that.  Or we can do a safeJsonParse function like the following that logs an error and returns an Internal Error back to the client.

```javascript
export function safeJsonParse(request, toParse) {
    try {
        return JSON.parse(toParse);
    } catch (error) {
        log.error('failed to parse json');
        return errors.InternalError;
    }
}
```